src/external/dftk_to_wannier90.jl

```diff
@@ -0,0 +1,415 @@
+    nn_kpts =  [parse.(Int,split(l,' ',keepempty = false))
+                for l in ln[i_nn_kpts[1]+2:i_nn_kpts[2]-1] ]
+    
+
+    # Extract projections block : NEEDED FOR WIN GUESSES
+    i_projs = findall(x-> endswith(x,"projections"),ln) # 1st and last line of the block   
+    @assert size(i_projs,1) == 2
+    raw_projs = split.(ln[i_projs[1]+1:i_projs[2]-1],' ',keepempty = false)
+
+    n_projs = parse(Int,only(popfirst!(raw_projs)))     # number of projections
+    @assert(n_projs == size(raw_projs,1)/2)
+
+    # Reshape so that one line gives all infos about one projection
+    raw_projs = reshape(raw_projs,(2,n_projs))
+    # Parse in the format proj = [ [[center],[l,mr,r],[z_axis],[x_axis], α], ... ]
+    projs = []
+    for j in 1:n_projs
+        center = parse.(Float64,raw_projs[1,j][1:3])
+        quantum_numbers = parse.(Int,raw_projs[1,j][4:end])
+        z_axis = parse.(Float64,raw_projs[2,j][1:3])
+        x_axis = parse.(Float64,raw_projs[2,j][4:6])
+        α = parse(Float64,raw_projs[2,j][7])
+        push!(projs, [center,quantum_numbers,z_axis,x_axis,α])
+    end
+    
+    nn_kpts,nn_num,projs    
+end
+
+
+
+"""
+Take scfres, the result of a self_consistent_field calculation and writes eigenvalues
+(in eV) in a format readable by Wannier90.
+"""
+function generate_eig_file(prefix::String, scfres)
+    open("$prefix.eig", "w") do f
+        for (k,e_k) in enumerate(scfres.eigenvalues)
+            for (n,e_kn) in enumerate(e_k) 
+                write(f,@sprintf("%3i  %3i   %25.18f \n",n,k,(1/DFTK.units.eV)*e_kn))
+            end
+        end
+    end
+    return nothing
+end
+
+
+@doc raw"""
+ Computes the matrix ``[M^{k,b}]_{m,n} = \langle u_{m,k} | u_{n,k+b} \rangle`` 
+ for given k, kpb = k+b.
+ 
+ Return in the format of a (n_bands^2, 2) matrix, where each line is one overlap, 
+ and columns are real and imaginary parts.
+
+ G_shift is the "shifting" vector, correction due to the periodicity conditions
+ imposed on k -> ψ_k. 
+ It is non zero if kpb is taken in another unit cell of the reciprocal lattice.
+ We use here that : ``u_{n(k + G_shift)}(r) = e^{-i*\langle G_shift,r \rangle} u_{nk}``
+"""
+function overlap_Mmn_k_kpb(basis::PlaneWaveBasis, ψ, ik::Integer, ikpb::Integer,
+                           n_bands::Integer;
+                           G_shift = [0,0,0])
+
+    Mkb = zeros(ComplexF64,(n_bands,n_bands))
+    # Search for common Fourier modes and their resp. indices in bloch states k and kpb
+    k = basis.kpoints[ik]; kpb = basis.kpoints[ikpb] #renaming for clarity
+    map_fourier_modes = [ (iGk,DFTK.index_G_vectors(basis, kpb, Gk+G_shift))
+                          for (iGk,Gk) in enumerate(G_vectors(k))
+                          if !isnothing(DFTK.index_G_vectors(basis,kpb,Gk+G_shift)) ]
+
+    # Compute overlaps
+    for n in 1:n_bands
+        for m in 1:n_bands
+            # Select the coefficient in right order
+            Gk_coeffs = @view ψ[ik][[i[1] for i in map_fourier_modes],m] 
```

Comment:
you're doing this `[i[1] for i in map_fourier_modes]` at each step, which is slow. Better to precompute it (which suggests that map_fourier_modes is not the right structure anyway)

Reply:
Yes I initially wanted to take the first and second column of a table but I don't know if it is feasible with Tuples.
Still I will take this out of the loop !